src/node_url.cc

```diff
@@ -705,10 +710,11 @@ namespace url {
             p++;
             continue;
           } else if (ch == ':' || (has_state_override && ch == kEOL)) {
```

Comment:
That is unfortunately true. We already deviate from the spec by storing `:` as part of the `scheme`, coupled by the fact that we delegate some responsibilities to the JS layer, it's hard to follow the spec word-for-word.